src/map.cpp

```diff
@@ -1312,6 +1312,11 @@ std::string map::name( const tripoint &p )
  return has_furn( p ) ? furn_at( p ).name : ter_at( p ).name;
 }
 
+std::string map::disp_name( const tripoint &p )
```

Comment:
This returns a temporary string, returns by `string_format`, when the functions returns, the reference would be invalid. However, the return of `map::name` could be a const reference, it refers to `ter_t::name` / `furn_t::name` which allocated once and stay valid until the game ends.